log-replication/src/main/java/org/corfudb/logreplication/infrastructure/CorfuInterClusterReplicationServerNode.java

```diff
@@ -0,0 +1,367 @@
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuInterruptedError;
+import org.corfudb.security.sasl.plaintext.PlainTextSaslNettyServer;
+import org.corfudb.security.tls.SslContextConstructor;
+import org.corfudb.util.GitRepositoryState;
+import org.corfudb.util.Version;
+
+import javax.annotation.Nonnull;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLException;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.regex.Pattern;
+
+@Slf4j
+public class CorfuInterClusterReplicationServerNode implements AutoCloseable {
+
+    @Getter
+    private final ServerContext serverContext;
+
+    @Getter
+    private final Map<Class, AbstractServer> serverMap;
+
+    @Getter
+    private final IServerRouter router;
+
+    // This flag makes the closing of the CorfuServer idempotent.
+    private final AtomicBoolean close;
+
+    private ChannelFuture bindFuture;
+
+    private boolean useNetty = true;
+
+    /**
+     * Corfu Server initialization.
+     *
+     * @param serverContext Initialized Server Context.
+     */
+    public CorfuInterClusterReplicationServerNode(@Nonnull ServerContext serverContext) {
+        this(serverContext,
+                ImmutableMap.<Class, AbstractServer>builder()
+                        .put(BaseServer.class, new BaseServer(serverContext))
+                        .put(LogReplicationServer.class, new LogReplicationServer(serverContext))
+                        .build()
+        );
+    }
+
+    /**
+     * Corfu Server initialization.
+     *
+     * @param serverContext Initialized Server Context.
+     * @param serverMap     Server Map with all components.
+     */
+    public CorfuInterClusterReplicationServerNode(@Nonnull ServerContext serverContext,
+                                                  @Nonnull Map<Class, AbstractServer> serverMap) {
+        this.serverContext = serverContext;
+        this.serverMap = serverMap;
+        this.close = new AtomicBoolean(false);
+
+        // If custom-transport is not specified by user, it will default to use Netty
+        if (serverContext.getServerConfig().get("--custom-transport") != null) {
+            useNetty = false;
+            router = new LogReplicationServerRouter(new ArrayList<>(serverMap.values()),
+                    Integer.parseInt((String) serverContext.getServerConfig().get("<port>")));
+            log.info("Corfu Replication Server initialized for CUSTOM transport.");
+        } else {
+            log.info("Corfu Replication Server initialized for NETTY transport.");
+            ImmutableList<AbstractServer> servers = ImmutableList.of((AbstractServer)serverMap.values());
```

Comment:
nit - the serverList retrieval is common for both netty and custom transport.  It can be moved outside the if-else blocks